spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -375,12 +376,67 @@ private static Iterator<ManifestFile> buildManifest(SerializableConfiguration co
    * @param sourceTableIdent an identifier of the source Spark table
    * @param targetTable an Iceberg table where to import the data
    * @param stagingDir a staging directory to store temporary manifest files
-   * @param partitionFilter only import partitions whose values match those in the map, can be partially defined
+   */
+  public static ImportSparkTableBuilder importSparkTableBuilder(SparkSession spark, TableIdentifier sourceTableIdent,
+                                                                Table targetTable, String stagingDir) {
+    return new ImportSparkTableBuilder(spark, sourceTableIdent, targetTable, stagingDir);
+  }
+
+  public static class ImportSparkTableBuilder {
```

Comment:
Do we need the builder here? I'm not sure this method is going to change much more in the future and we tend to want to limit Builders to those apis we think will evolve a lot in the future.

Reply:
It's been removed.

Because this method takes a lot of parameters and can not set default parameter values like scala, so i added a Builder. :sweat_smile:

Reply:
Also, I had to create multiple methods to distinguish checkDuplicateFiles from skipOnError.  It can not specify parameter name when values are passed in, as Scala does. 😂